eclair-core/src/main/scala/fr/acinq/eclair/crypto/KeyManager.scala

```diff
@@ -110,6 +110,16 @@ trait KeyManager {
    *         private key, bitcoinSig is the signature of the channel announcement with our funding private key
    */
   def signChannelAnnouncement(fundingKeyPath: DeterministicWallet.KeyPath, chainHash: ByteVector32, shortChannelId: ShortChannelId, remoteNodeId: PublicKey, remoteFundingKey: PublicKey, features: Features): (ByteVector64, ByteVector64)
+
+  /**
+   * Sign a digest, primarily used to prove ownership of the current node
+   *
+   * @param digest     SHA256 digest
+   * @param privateKey private key to sign with, default the one from the current node
+   * @return a (signature, recoveryId) pair. signature is a signature of the digest parameter generated with the
+             private key given in parameter. recoveryId is the corresponding recoveryId of the signature
```

Comment:
I would take this opportunity to explain the recoveryId in more details:

```
When recovering a public key from an ECDSA signature for secp256k1, there are 4 possible matching curve points that can be found. The recoveryId identifies which of these points is the correct one to speed up recovery (otherwise we would need to check the signature's validity for each of these points to find the correct one).
```

Reply:
Done